pkg/sdk/testint/grants_integration_test.go

```diff
@@ -935,41 +1059,246 @@ func TestInt_GrantOwnership(t *testing.T) {
+		require.NoError(t, err)
+
+		// Use a previously prepared role to create a pipe and grant MONITOR + OPERATE to the previously used role (ACCOUNTADMIN).
+		usePreviousRole := useRole(t, client, pipeRole.Name)
+
+		pipe, pipeCleanup := createPipe(t, client, testDb(t), testSchema(t), random.AlphaN(20), copyStatement)
+		t.Cleanup(func() {
+			usePreviousRole = useRole(t, client, role.Name)
+			pipeCleanup()
+			usePreviousRole()
+		})
+
+		// Grant MONITOR and OPERATE privileges to the role.
+		makeAccountRoleOperableOnPipe(t, previousRole, pipe)
+
+		usePreviousRole()
+
+		err = client.Pipes.Alter(ctx, pipe.ID(), &sdk.AlterPipeOptions{
+			Set: &sdk.PipeSet{
+				PipeExecutionPaused: sdk.Bool(false),
+			},
+		})
+		require.NoError(t, err)
+
+		pipeExecutionState, err := client.SystemFunctions.PipeStatus(pipe.ID())
+		require.NoError(t, err)
+		require.Equal(t, sdk.RunningPipeExecutionState, pipeExecutionState)
+
+		err = client.Grants.GrantOwnership(
+			ctx,
+			ownershipGrantOnPipe(pipe),
+			sdk.OwnershipGrantTo{
+				AccountRoleName: sdk.Pointer(role.ID()),
+			},
+			&sdk.GrantOwnershipOptions{
+				CurrentGrants: &sdk.OwnershipCurrentGrants{
+					OutboundPrivileges: sdk.Copy, // With copy, we'll be able to resume the pipe after ownership transfer
+				},
+			},
+		)
+		require.NoError(t, err)
+		checkOwnershipOnObjectToRole(t, ownershipGrantOnPipe(pipe), role.ID().Name())
+
+		usePreviousRole()
 
 		pipeExecutionState, err = client.SystemFunctions.PipeStatus(pipe.ID())
 		require.NoError(t, err)
 		require.Equal(t, sdk.RunningPipeExecutionState, pipeExecutionState)
 	})
 
-	t.Run("on all pipes", func(t *testing.T) {
-		table, tableCleanup := createTable(t, itc.client, testDb(t), testSchema(t))
-		t.Cleanup(tableCleanup)
+	t.Run("on pipe - with neither ownership nor operate", func(t *testing.T) {
+		role, roleCleanup := createRoleGrantedToCurrentUser(t, client)
+		t.Cleanup(roleCleanup)
 
-		stage, stageCleanup := createStage(t, itc.client, sdk.NewSchemaObjectIdentifier(testDb(t).Name, testSchema(t).Name, random.AlphaN(20)))
-		t.Cleanup(stageCleanup)
+		pipeRole, pipeRoleCleanup := createRoleGrantedToCurrentUser(t, client)
+		t.Cleanup(pipeRoleCleanup)
+
+		// Role needs usage on the database and schema to later be able to remove pipe in the cleanup
+		grantDatabaseAndSchemaUsage(t, role)
+		// grantPipeRole grants the necessary privileges to a role to be able to create pipe
+		grantPipeRole(t, pipeRole, table, stage)
+
+		// Use a previously prepared role to create a pipe and grant MONITOR + OPERATE to the previously used role (ACCOUNTADMIN).
+		usePreviousRole := useRole(t, client, pipeRole.Name)
 
-		copyStatement := createPipeCopyStatement(t, table, stage)
+		pipe, pipeCleanup := createPipe(t, client, testDb(t), testSchema(t), random.AlphaN(20), copyStatement)
+		t.Cleanup(func() {
+			usePreviousRole = useRole(t, client, pipeRole.Name)
+			pipeCleanup()
+			usePreviousRole()
```

Comment:
what if pipeCleanup fails? we never get to usePReviousROle, should we defer it right after `usePreviousRole = useRole(t, client, pipeRole.Name)`? The same goes for other tests too.